File: 912-Sort an Array.py
class Solution(object):
    def sortArray(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
# merge sort
        def mergeSort(arr):
            if len(arr) < 2:
                return arr
            
            mid = len(arr)//2
            leftArr = mergeSort(arr[:mid])
            rightArr = mergeSort(arr[mid:])
            i = j = k = 0
            
            while i<len(leftArr) and j<len(rightArr):
                if leftArr[i] < rightArr[j]:
                    arr[k] = leftArr[i]
                    i+=1
                else:
                    arr[k] = rightArr[j]
                    j+=1
                k+=1
                
            while i<len(leftArr):
                arr[k] = leftArr[i]
                i+=1
                k+=1
                
            while j<len(rightArr):
                arr[k] = rightArr[j]
                j+=1
                k+=1
                
            return arr
        
        return mergeSort(nums)

File: JavaScript/912-Sort an Array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArray = function(nums) {
    let mergeSort = (arr) =>{
        if (arr.length < 2) return arr;
        const mid = Math.floor(arr.length/2);
        let leftArr = mergeSort(arr.slice(0, mid));
        let rightArr = mergeSort(arr.slice(mid));
        
        let i=0, j=0, k=0;
        while (i<leftArr.length && j<rightArr.length){
            if (leftArr[i] < rightArr[j]){
                arr[k] = leftArr[i];
                i++;
            }else{
                arr[k] = rightArr[j];
                j++;
            }
            k++;
        }
        while (i<leftArr.length){
            arr[k] = leftArr[i];
            i++;k++;
        }
        while (j<rightArr.length){
            arr[k] = rightArr[j];
            j++;k++;
        }        
        return arr;
    }
    return mergeSort(nums);
};